packages/bcp/src/cosmwasmcodec.spec.ts

```diff
@@ -1,64 +1,128 @@
-      );
-      expect(cosmWasmCodec.isValidAddress("cosmosvaloper17mggn4znyeyg25wd7498qxl7r2jhgue8u4qjcq")).toEqual(
-        true,
-      );
+      expect(codec.isValidAddress("cosmos1pkptre7fdkl6gfrzlesjjvhxhlc3r4gmmk8rs6")).toEqual(true);
+      expect(codec.isValidAddress("cosmosvalcons10q82zkzzmaku5lazhsvxv7hsg4ntpuhdwadmss")).toEqual(true);
+      expect(codec.isValidAddress("cosmosvaloper17mggn4znyeyg25wd7498qxl7r2jhgue8u4qjcq")).toEqual(true);
     });
 
     it("rejects invalid addresses", () => {
       // Bad size
-      expect(cosmWasmCodec.isValidAddress("cosmos10q82zkzzmaku5lazhsvxv7hsg4ntpuhh8289f")).toEqual(false);
+      expect(codec.isValidAddress("cosmos10q82zkzzmaku5lazhsvxv7hsg4ntpuhh8289f")).toEqual(false);
       // Bad checksum
-      expect(cosmWasmCodec.isValidAddress("cosmos1pkptre7fdkl6gfrzlesjjvhxhlc3r4gmmk8rs7")).toEqual(false);
+      expect(codec.isValidAddress("cosmos1pkptre7fdkl6gfrzlesjjvhxhlc3r4gmmk8rs7")).toEqual(false);
       // Bad prefix
-      expect(cosmWasmCodec.isValidAddress("cosmot10q82zkzzmaku5lazhsvxv7hsg4ntpuhd8j5266")).toEqual(false);
+      expect(codec.isValidAddress("cosmot10q82zkzzmaku5lazhsvxv7hsg4ntpuhd8j5266")).toEqual(false);
     });
   });
 
-  it("properly generates bytes to sign", () => {
-    const expected = {
-      bytes: toUtf8(
-        '{"account_number":"0","chain_id":"cosmoshub-3","fee":{"amount":[{"amount":"2500","denom":"uatom"}],"gas":"100000"},"memo":"","msgs":[{"type":"cosmos-sdk/MsgSend","value":{"amount":[{"amount":"35997500","denom":"uatom"}],"from_address":"cosmos1txqfn5jmcts0x0q7krdxj8tgf98tj0965vqlmq","to_address":"cosmos1nynns8ex9fq6sjjfj8k79ymkdz4sqth06xexae"}}],"sequence":"99"}',
-      ),
-      prehashType: PrehashType.Sha256,
-    };
-    const bytesToSign = cosmWasmCodec.bytesToSign(sendTxJson, nonce);
+  describe("bytesToSign", () => {
+    it("works for SendTransaction via bank module", () => {
+      const expected = {
+        bytes: toUtf8(
+          '{"account_number":"0","chain_id":"cosmoshub-3","fee":{"amount":[{"amount":"2500","denom":"uatom"}],"gas":"100000"},"memo":"","msgs":[{"type":"cosmos-sdk/MsgSend","value":{"amount":[{"amount":"35997500","denom":"uatom"}],"from_address":"cosmos1txqfn5jmcts0x0q7krdxj8tgf98tj0965vqlmq","to_address":"cosmos1nynns8ex9fq6sjjfj8k79ymkdz4sqth06xexae"}}],"sequence":"99"}',
+        ),
+        prehashType: PrehashType.Sha256,
+      };
+      expect(codec.bytesToSign(sendTxJson, nonce)).toEqual(expected);
+    });
 
-    expect(bytesToSign).toEqual(expected);
-  });
+    it("works for ERC20 send", () => {
+      const bashSendTx: SendTransaction = {
+        kind: "bcp/send",
+        chainId: chainId,
+        sender: "cosmos1txqfn5jmcts0x0q7krdxj8tgf98tj0965vqlmq" as Address,
+        recipient: "cosmos1dddd" as Address,
+        memo: "My first BASH payment",
+        amount: {
+          fractionalDigits: 0,
+          quantity: "345",
+          tokenTicker: "BASH" as TokenTicker,
+        },
+        fee: {
```

Comment:
wow, almost 1/3 of this tx is just setting the fee and memo, which we would like to make automatic. Maybe use `defaultFee` for the tests. I definitely need to think about how to configure this better in production.

(I'm really looking at the API and what we can simplify in a basic API, without removing the power of the full API - I really think we need some wrapper class which makes lots of assumptions for intro devs, then they can pull off the wrapper and grab all the fields when they need it)

Reply:
For those kind of encoder tests I prefer to be specific to allow seeing what is going on and to reduce dependencies as much as possible. The example in "can send ERC20 tokens" at least removes the fee noise.

Reply:
Good call. This is encoding. Totally right, j was thinking usage, and it did seem less noisy